joomla/joomla-cms
# Proper alias transliteration based on the content language

Currently if "Unicode Aliases" are disabled in global configuration, transliteration is made based on the current language used in administration area. This pull request will make all core extensions aliases to be transliterated based on the actual resource ( article, category, banner etc. ) language . If "All" is selected as the resource language then the site's default language will be used as a fallback.
#### Summary of Changes

Since all core extensions are using the JApplicationHelper::stringURLSafe which calls the JFilterOutput::stringURLSafe both of them are changed with a second extra argument holding the language variable. All the core table classes which are calling ApplicationHelper::stringURLSafe for generating the alias are also changed to provide the language of the resource.
#### Testing Instructions
1. Create a site with two languages ( Let's say English and Greek ).
2. Set the (backend) language to English.
3. Disable "Unicode aliases" in global configuration.
4. Create an article in Greek language (using a title in Greek and also remember to select "Greek" in the language field ) .

The generated alias now should be transliterated correctly. Without this pull request the result would be a date based alias.
